src/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using Dynamo.Logging;
+using Dynamo.Wpf.Interfaces;
+using ImageMagick;
+using Newtonsoft.Json;
+
+namespace NodeDocumentationMarkdownGenerator
+{
+    internal static class MarkdownHandler
+    {
+        /// <summary>
+        /// Creates markdown files from the given collection of fileInfos
+        /// </summary>
+        /// <param name="fileInfos">Collection of files to create</param>
+        /// <param name="outputDir">Folder path where files should be created</param>
+        /// <param name="overWrite">if true, files in outputDir will be overwritten</param>
+        /// <param name="logger">logger instance</param>
+        /// <param name="compressImages">if true images matched from dictionary will be compressed (if possible)</param>
+        /// <param name="dictionaryPath">path to dictionary json file</param>
+        /// <param name="layoutSpec">path to layout spec json</param>
+        internal static void CreateMdFilesFromFileNames(List<MdFileInfo> fileInfos, string outputDir, bool overWrite, ILogger logger, bool compressImages = false, string dictionaryPath = null, string layoutSpec = null)
+        {
+            ImageOptimizer optimizer = null;
+            LayoutSpecification spec = null;
+            List<DynamoDictionaryEntry> dictEntrys = null;
+            string examplesDirectory = "";
+            if (!string.IsNullOrEmpty(dictionaryPath) &&
+                    File.Exists(dictionaryPath))
+            {
+                var dictionaryJson = File.ReadAllText(dictionaryPath);
+                dictEntrys = JsonConvert.DeserializeObject<List<DynamoDictionaryEntry>>(dictionaryJson);
+                var mainDirectory = new FileInfo(dictionaryPath).Directory;
+                examplesDirectory = Path.Combine(mainDirectory.FullName, "EXAMPLES");
+
+                if (compressImages)
+                {
+                    optimizer = new ImageOptimizer();
+                    optimizer.OptimalCompression = true;
+                }
```

Comment:
should the image compression only happen if the dictionary path is provided ? 
ie, should we not also optimise images for ZT authors if they set it to true ?

Reply:
For now i removed the option for a dictionary path for package authors. The sturcuture of the dictionary json is.... well, not great, so if that sturcture is to be used by 3rd party devs it probably needs a little work